swarm/pss/pss_test.go

```diff
@@ -61,431 +66,656 @@ func init() {
+	if ps.isSelfPossibleRecipient(pssmsg) {
+		t.Fatalf("isSelfPossibleRecipient true but %x != %x", remoteaddr[:8], localaddr[:8])
 	}
 
-	if !bytes.Equal(digest[:], proofbytes) {
-		t.Fatalf("digest - got: %x, expected: %x", digest, proofbytes)
+	// 8 first bytes same
+	copy(remoteaddr[:4], localaddr[:4])
+	if ps.isSelfRecipient(pssmsg) {
+		t.Fatalf("isSelfRecipient true but %x != %x", remoteaddr, localaddr)
 	}
-
-	if digest == digesttwo {
-		t.Fatalf("different msgs return same crc: %d", digesttwo)
+	if !ps.isSelfPossibleRecipient(pssmsg) {
+		t.Fatalf("isSelfPossibleRecipient false but %x == %x", remoteaddr[:8], localaddr[:8])
 	}
 
-	// check the sender cache
-	err = ps.addFwdCacheSender(fwdaddr.Over(), digest)
-	if err != nil {
-		t.Fatalf("write to pss sender cache failed: %v", err)
+	// all bytes same
+	pssmsg.To = localaddr
+	if !ps.isSelfRecipient(pssmsg) {
+		t.Fatalf("isSelfRecipient false but %x == %x", remoteaddr, localaddr)
 	}
-
-	if !ps.checkFwdCache(fwdaddr.Over(), digest) {
-		t.Fatalf("message %v should have SENDER record in cache but checkCache returned false", msg)
+	if !ps.isSelfPossibleRecipient(pssmsg) {
+		t.Fatalf("isSelfPossibleRecipient false but %x == %x", remoteaddr[:8], localaddr[:8])
 	}
+}
 
-	if ps.checkFwdCache(fwdaddr.Over(), digesttwo) {
-		t.Fatalf("message %v should NOT have SENDER record in cache but checkCache returned true", msgtwo)
-	}
+// tests:
+// sets public key for peer
+// set an outgoing symmetric key for peer
+// generate own symmetric key for incoming message from peer
+func TestKeys(t *testing.T) {
+	outkey := make([]byte, 32) // whisper aes symkeys are 32, if that changes this test will break
 
-	// check the expire cache
-	err = ps.addFwdCacheExpire(digest)
+	// make our key and init pss with it
+	ourkeys, err := wapi.NewKeyPair()
+	theirkeys, err := wapi.NewKeyPair()
 	if err != nil {
-		t.Fatalf("write to pss expire cache failed: %v", err)
+		t.Fatalf("create key fail")
 	}
+	ourprivkey, err := w.GetPrivateKey(ourkeys)
+	theirprivkey, err := w.GetPrivateKey(theirkeys)
+	ps := NewTestPss(ourprivkey)
 
-	if !ps.checkFwdCache(nil, digest) {
-		t.Fatalf("message %v should have EXPIRE record in cache but checkCache returned false", msg)
-	}
+	// set up peer with mock address, mapped to mocked publicaddress and with mocked symkey
+	addr := network.RandomAddr().Over()
+	topic := whisper.BytesToTopic([]byte("foo:42"))
+	ps.SetOutgoingPublicKey(pot.NewAddressFromBytes(addr), topic, &theirprivkey.PublicKey)
+	copy(outkey[:16], addr[16:])
+	copy(outkey[16:], addr[:16])
+	outkeyid, err := ps.SetOutgoingSymmetricKey(pot.NewAddressFromBytes(addr), topic, outkey)
 
-	if ps.checkFwdCache(nil, digesttwo) {
-		t.Fatalf("message %v should NOT have EXPIRE record in cache but checkCache returned true", msgtwo)
-	}
+	// make a symmetric key that we will send to peer for encrypting messages to us
+	inkeyid, err := ps.GenerateIncomingSymmetricKey(pot.NewAddressFromBytes(addr), topic)
```

Comment:
There are lot's of unchecked errors in these tests which all should be asserted to be not nil.